packages/go_router/doc/configuration.md

```diff
@@ -0,0 +1,140 @@
+Create a GoRouter configuration by calling the [GoRouter][] constructor and
+providing list of [GoRoute][] objects:
+
+```dart
+GoRouter(
+  routes: [
+    GoRoute(
+      path: '/',
+      builder: (context, state) => const Page1Screen(),
+    ),
+    GoRoute(
+      path: '/page2',
+      builder: (context, state) => const Page2Screen(),
+    ),
+  ],
+);
+```
+
+# GoRoute
+To configure a GoRoute, a path template and builder must be provided. Specify a
+path template to handle by providing a `path` parameter, and a builder by
+providing either the `builder` or `pageBuilder` parameter:
+
+```dart
+GoRoute(
+  path: '/users/:userId',
+  builder: (context, state) => const UserScreen(),
+),
+```
+
+# Parameters
+To specify a path parameter, prefix a path segment with a `:` character,
+followed by a unique name, for example, `:userId`. You can access the value of
+the parameter by accessing it through the [GoRouterState][] object provided to
+the builder callback:
+
+```dart
+GoRoute(
+  path: '/users/:userId',
+  builder: (context, state) => const UserScreen(id: state.params['userId']),
+),
+```
+
+Similarly, to access a query parameter, use [GoRouterState][]:
```

Comment:
Should we explain what query parameters are? This content may be confusing for newbies that aren't familiar with them.